### PR TITLE
feat(Slider): redesign API to make more usable

### PR DESCRIPTION
**Pull request by Arucard89**
Make API more flexible and convenient
UXRFC-503

### CODE REVIEW

File: src/components/Slider/BaseSlider/BaseSlider.scss [hunk 1]
```diff
@@ -251,6 +251,10 @@ $rail-xl-size: 6px;
         }
     }
 
+    &_noMarks {
```
Comment (amje) [inline src/components/Slider/BaseSlider/BaseSlider.scss:254]:
```suggestion
    &_no-marks {
```

Prefer not using camelCase in CSS

File: src/components/Slider/README.md [hunk 2]
```diff
@@ -190,96 +190,143 @@ LANDING_BLOCK-->
...
+| [marksFormat](#marks)                       | Formatter for the mark's displayed value                                                                                                                                                        |             `(value: number) => string`              |         |
+| [max](#min-and-max-value)                   | The maximum value of the component.                                                                                                                                                             |                       `number`                       |  `100`  |
+| [min](#min-and-max-value)                   | The maximum value of the component.                                                                                                                                                             |                       `number`                       |   `0`   |
```
Comment (amje) [inline src/components/Slider/README.md:321]:
```suggestion
| [min](#min-and-max-value)                   | The minimum value of the component.                                                                                                                                                             |                       `number`                       |   `0`   |
```

File: src/components/Slider/Slider.scss [hunk 3]
```diff
@@ -32,9 +32,9 @@ $block: '.#{variables.$ns}slider';
         height: calc(
             var(--g-text-body-1-line-height) + 6px + #{slider-variables.$slider-tooltip-offset}
         );
-        display: none;
-        &_hasTooltip {
-            display: block;
+        display: block;
+        &_tooltipDisplay_off {
```
Comment (amje) [inline src/components/Slider/Slider.scss:36]:
```suggestion
        &_tooltip-display_off {
```

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Slider/Slider.scss (hunk 3 above)
Comment (Arucard89) [reply src/components/Slider/Slider.scss]: bee69a167607e75a50419790e0eeae46bec78ba5

File: src/components/Slider/README.md (hunk 2 above)
Comment (Arucard89) [reply src/components/Slider/README.md]: bee69a167607e75a50419790e0eeae46bec78ba5

**Review by amje — APPROVED** [review]

**Pull request merged**